client/views/view_economics.cpp

```diff
@@ -90,32 +91,37 @@ void eco_report::update_report()
     ui.eco_widget->insertRow(i);
     for (j = 0; j < 6; j++) {
       item = new QTableWidgetItem;
+      QLabel *lbl = new QLabel;
```

Comment:
You leak the label when `j != 0` (memory leak): it is neither deleted nor used. Simply create it when getting the sprite.

Reply:
How is that different than the `QTableWidgetItem`?